src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Windows/FindPal.cs

```diff
@@ -0,0 +1,405 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using System.Numerics;
+using System.Runtime.InteropServices;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using Internal.Cryptography.Pal.Native;
+using Internal.NativeCrypto;
+
+namespace Internal.Cryptography.Pal
+{
+    internal partial class FindPal : IFindPal
+    {
+        private readonly StorePal _storePal;
+        private readonly X509Certificate2Collection _copyTo;
+        private readonly bool _validOnly;
+
+        internal FindPal(X509Certificate2Collection findFrom, X509Certificate2Collection copyTo, bool validOnly)
+        {
+            _storePal = (StorePal)StorePal.LinkFromCertificateCollection(findFrom);
+            _copyTo = copyTo;
+            _validOnly = validOnly;
+        }
+
+        internal static IFindPal OpenPal(X509Certificate2Collection findFrom, X509Certificate2Collection copyTo, bool validOnly)
+        {
+            return new FindPal(findFrom, copyTo, validOnly);
+        }
+
+        public string NormalizeOid(string maybeOid, OidGroup expectedGroup)
+        {
+            string oidValue = OidInfo.FindOidInfo(CryptOidInfoKeyType.CRYPT_OID_INFO_NAME_KEY, maybeOid, expectedGroup, fallBackToAllGroups: true).OID;
+
+            if (oidValue == null)
+            {
+                oidValue = maybeOid;
+                ValidateOidValue(oidValue);
+            }
+
+            return oidValue;
+        }
+
+        public unsafe void FindByThumbprint(byte[] thumbPrint)
+        {
+            fixed (byte* pThumbPrint = thumbPrint)
+            {
+                CRYPTOAPI_BLOB blob = new CRYPTOAPI_BLOB(thumbPrint.Length, pThumbPrint);
+                FindCore(CertFindType.CERT_FIND_HASH, &blob);
+            }
+        }
+
+        public unsafe void FindBySubjectName(string subjectName)
+        {
+            fixed (char* pSubjectName = subjectName)
+            {
+                FindCore(CertFindType.CERT_FIND_SUBJECT_STR, pSubjectName);
+            }
+        }
+
+        public void FindBySubjectDistinguishedName(string subjectDistinguishedName)
+        {
+            FindCore(
+                delegate(SafeCertContextHandle pCertContext)
```

Comment:
> Why the anonymous method syntax here when you used the lambda syntax elsewhere

These were just moved from the file where they were to this new location during the PAL restructuring.  They were declared delegate in the desktop code, probably because they predated lambdas by about 3 major releases :).